contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/problem/ProblemDetailsMockMvcMatchers.java

```diff
@@ -0,0 +1,189 @@
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class ProblemDetailsMockMvcMatchers {
+
+    private final static ObjectMapper objectMapper = new ObjectMapper();
+
+    static {
+        objectMapper.addMixIn(ProblemDetail.class, ProblemDetailJacksonMixin.class);
+    }
+
+    public static ProblemDetailsMatcher problemDetails() {
+        return new ProblemDetailsMatcher();
+    }
+
+    public static ValidationConstraintViolationMatcher validationConstraintViolation() {
+        return new ValidationConstraintViolationMatcher(List.of());
+    }
+
+    @With
+    @AllArgsConstructor
+    public static class ProblemDetailsMatcher implements ResultMatcher {
+
+        private final static String SENTINEL = "\0";
+        private final String type;
+        private final String title;
+        private final HttpStatusCode statusCode;
+
+        public ProblemDetailsMatcher() {
+            this(SENTINEL, SENTINEL, null);
+        }
+
+        ProblemDetail readProblemDetail(MvcResult result) throws IOException {
+            assertThat(result.getResponse().getContentType())
+                    .as("response Content-Type")
+                    .isEqualTo("application/problem+json");
+
+            var problemDetails = objectMapper.reader()
+                    .readValue(result.getResponse().getContentAsByteArray(), ProblemDetail.class);
+
+            if (statusCode != null) {
+                assertThat(result.getResponse().getStatus())
+                        .as("response status code")
+                        .isEqualTo(statusCode.value());
+                assertThat(problemDetails)
+                        .extracting(ProblemDetail::getStatus)
+                        .as("problem status")
+                        .isEqualTo(statusCode.value());
+            }
+
+            if (!Objects.equals(type, SENTINEL)) {
+                assertThat(problemDetails)
+                        .extracting(ProblemDetail::getType)
+                        .extracting(URI::toString)
+                        .as("problem type")
+                        .isEqualTo(type);
+            }
+            if (!Objects.equals(title, SENTINEL)) {
+                assertThat(problemDetails)
+                        .extracting(ProblemDetail::getTitle)
+                        .as("problem title")
+                        .isEqualTo(title);
+            }
+
+            return problemDetails;
+        }
+
+        @Override
+        public void match(MvcResult result) throws Exception {
+            readProblemDetail(result);
+        }
+    }
+
+    @AllArgsConstructor
+    public static class ValidationConstraintViolationMatcher implements ResultMatcher {
+
+        private final static ProblemDetailsMatcher PROBLEM_DETAILS_MATCHER = new ProblemDetailsMatcher()
+                .withStatusCode(HttpStatus.BAD_REQUEST)
+                .withType("https://contentgrid.cloud/problems/integrity/validation-constraint-violation");
```

Comment:
Shouldn't we extract this url (or at least the base) into a dedicated class ?

Reply:
The URL should not change, because it is the identity of the problem, right?

Reply:
It might be nice to be able to refer to it with IDE autocomplete, rather than having to carefully copy-paste it all the time.

Reply:
:point_up:

Reply:
Well, in the test class, the prefix is there in a constant, which is the one you would mostly use.

Reply:
:shrug: